traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm

```diff
@@ -2308,13 +2314,22 @@ sub parent_dot_config { #fix qstring - should be ignore for quika
 		}
 		my %seen;
 		@parent_info = grep { !$seen{$_}++ } @parent_info;
-		my $parents = 'parent="' . join( '', sort @parent_info ) . '"';
-		my $secparents = '';
 		if ( scalar @secondary_parent_info > 0 ) {
 			my %seen;
 			@secondary_parent_info = grep { !$seen{$_}++ } @secondary_parent_info;
-			$secparents = 'secondary_parent="' . join( '', sort @secondary_parent_info ) . '"';
 		}
+		#If the ats version supports it, put secondary parents into secondary parent group.
+		#This will ensure that secondary parents will be unused unless all hosts in the primary group are unavailable.
+		my $parents = '';
+		my $secparents = '';
+		if ( $ats_major_version >= 6 && @secondary_parent_info > 0 ) {
+			$parents = 'parent="' . join( '', sort @parent_info ) . '"';
+			$secparents = ' secondary_parent="' . join( '', sort @secondary_parent_info ) . '"';
+		}
+		else {
```

Comment:
Might need the `@secondary_parent_info > 0` conditional here as well. Can you check this out?

Reply:
Nope.  It's only in the first conditional because I don't want it to modify $secparents unless there are actually secondaries.